custom_components/ramses_cc/sensor.py

```diff
@@ -127,309 +147,305 @@ async def async_setup_platform(
+class RamsesSensor(RamsesEntity, SensorEntity):
     """Representation of a generic sensor."""
 
-    # Strictly: fan_info, oem_code are not a measurements
-    # _attr_native_unit_of_measurement: str = PERCENTAGE  # most common
-    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT  # all but 2
+    entity_description: RamsesSensorEntityDescription
 
     def __init__(
         self,
-        broker: RamsesBroker,  # ramses_cc broker
-        device,  # ramses_rf device
-        state_attr,  # key of attr_dict +/- _ot suffix
-        device_class=None,  # attr_dict value
-        device_units=None,  # attr_dict value
-        state_class=SensorStateClass.MEASUREMENT,  # attr_dict value, maybe None
-        **kwargs,  # leftover attr_dict values, incl. 'name'
-    ) -> None:
-        """Initialize a sensor."""
-        _LOGGER.info("Found %r: %s", device, state_attr)
-        super().__init__(broker, device, state_attr, device_class=device_class)
-
-        self._attr_native_unit_of_measurement = device_units
-        self._attr_state_class = state_class
-
-    @property
-    def native_value(self) -> Any | None:  # int or float
-        """Return the state of the sensor."""
-        state = getattr(self._device, self._state_attr)
-        if self._attr_native_unit_of_measurement == PERCENTAGE:
-            return None if state is None else state * 100
-        return state
-
-
-class RamsesCo2Sensor(RamsesSensor):
-    """Representation of a generic sensor."""
-
-    @callback
-    def async_put_co2_level(self, co2_level: int = None, **kwargs) -> None:
-        """Set the state of the Sensor."""
-        self._device.co2_level = co2_level
-
-
-class RamsesIndoorHumidity(RamsesSensor):
-    """Representation of a generic sensor."""
-
-    @callback
-    def async_put_indoor_humidity(
-        self, indoor_humidity: float = None, **kwargs
+        broker: RamsesBroker,
+        device: RamsesRFEntity,
+        entity_description: RamsesEntityDescription,
     ) -> None:
-        """Set the state of the Sensor."""
-        self._device.indoor_humidity = indoor_humidity / 100
+        """Initialize the sensor."""
+        super().__init__(broker, device, entity_description)
 
-
-class RamsesHeatDemand(RamsesSensor):
-    """Representation of a heat demand sensor."""
+        self.entity_id = ENTITY_ID_FORMAT.format(
+            f"{device.id}_{entity_description.attr}"
+        )
+        self._attr_unique_id = f"{device.id}-{entity_description.key}"
 
     @property
-    def icon(self) -> str:
-        """Return the icon to use in the frontend, if any."""
-        return "mdi:radiator-off" if self.state == 0 else "mdi:radiator"
-
-
-class RamsesModLevel(RamsesSensor):
-    """Representation of a heat demand sensor."""
+    def available(self) -> bool:
+        """Return True if the entity is available."""
+        return self.state is not None
 
     @property
-    def extra_state_attributes(self) -> dict[str, Any]:
```

Comment:
I believe I'm correct in thinking that these are now all exposed as individual sensors with a global preference for the manner in which they are sourced.